catalogue/webapp/pages/search/index.tsx

```diff
@@ -31,16 +108,96 @@ export const getServerSideProps: GetServerSideProps<Props | AppErrorProps> =
       return { notFound: true };
     }
 
-    return {
-      props: removeUndefinedProps({
-        serverData,
-        // TODO Harrison to explore what properties we'd want here
-        pageview: {
-          name: 'search',
-          properties: {},
+    const query = context.query;
+    const params = fromQuery(query);
+
+    const defaultProps = removeUndefinedProps({
+      serverData,
+      query,
+
+      // TODO Harrison to explore what properties we'd want here
+      pageview: {
+        name: 'stories',
+        properties: {},
+      },
+    });
+
+    // Stop here if no query has been entered
+    if (!params.query) {
+      return {
+        props: defaultProps,
+      };
+    }
+
+    try {
+      // Helper function as this is repeated across all queries
+      const setData = (categoryName, fetchedData): FetchedDataProps => {
+        // An error shouldn't stop the other results from displaying
+        if (fetchedData.type === 'Error') {
+          console.error(fetchedData.label + ': Error fetching ' + categoryName);
+        }
+
+        return fetchedData && fetchedData.type !== 'Error'
+          ? fetchedData.results
+          : undefined;
+      };
+
+      /*
+       * Stories
+       */
+      const storiesFetch = await getStories({
+        query,
+        pageSize: 3,
+      });
+      const stories = setData('stories', storiesFetch) as Story[] | undefined;
+
+      /*
+       * Works
+       */
+      const _worksQueryType = getCookie('_queryType') as string | undefined;
+      const worksApiProps = {
+        ...params,
+        _queryType: _worksQueryType,
+      };
+      const worksFetch = await getWorks({
+        params: worksApiProps,
+        pageSize: 3,
+        toggles: serverData.toggles,
+      });
+      const works = setData('works', worksFetch) as Work[] | undefined;
+
+      /*
+       * Images
+       */
+      const imagesFetch = await getImages({
+        params,
+        toggles: serverData.toggles,
+        pageSize: 3,
+      });
+      const images = setData('images', imagesFetch) as Image[] | undefined;
+
+      // But if all three queries fail, return an error page
+      if (
+        imagesFetch.type === 'Error' &&
+        worksFetch.type === 'Error' &&
+        storiesFetch.type === 'Error'
+      ) {
+        return appError(context, 500, 'Search results error');
+      }
+      /*
+       * Return results or undefined for each category
+       */
```

Comment:
I'm not sure this comment is adding anything tbh.

Reply:
And it's not relevant anymore! Deleting :)